### PR TITLE
CSADigitizer: fix global time

### LINKED ISSUE #28: CSADigitizer: Wrong time when setting ToA binsize (?)
I've looked a little bit in the code of the ToA / ToT measurement, and something seems off to me. I'm not sure if this is just me being stupid or if there is actually something wrong.

Looking at: https://github.com/allpix-squared/allpix-squared/blob/2005e9f71f3944c0eecc9b1c54080df0110bd72e/src/modules/CSADigitizer/CSADigitizerModule.cpp#L342

We seem to add the local time, the global time, and the charge in the vector. "charge" in this case can also mean ToT signal (or rather TS2, I'll explain later). Everything is fine there. However, time can also mean the ToA signal in clock cycles. That in itself is also fine, if we know that. However, adding `time` to `pixel_charge.getGlobalTime()` doesn't make sense in that case. We add clock cycles to something which is in nanoseconds. Am I missing something, or is this wrong?

A fix could be multiplying the `time` with the bin size here: https://github.com/allpix-squared/allpix-squared/blob/2005e9f71f3944c0eecc9b1c54080df0110bd72e/src/modules/CSADigitizer/CSADigitizerModule.cpp#L321

Related to this:
I realized that the signal only is the "ToT", but that actually isn't the time over threshold, but rather only the second timestamp (that's why I prefer TS2 / TS1 to avoid misconceptions). A proper ToT would be `TS2-TS1`. Of course, this would mean that we would need to add some code that handles different clock sizes. One idea would be simply to do `TS2_signal * TS2_clk - ToA_signal * ToA_clk`, which then wouldn't return the signal in clock cycles but the resulting time from it.

I see that this would drastically change the way how this would work, and this can be relatively easy achieved by analysis after the run. Let me know what you think about this.

### PR DESCRIPTION
**Pull request by stephanlachnit**
Closes #28.

### CODE REVIEW

**Comment by simonspa** [comment]
Thanks! Would you mind extending the README to clearly state what the fields of the `PixelHit` are depending on the ToT and ToA settings?

**Comment by stephanlachnit** [comment]
Yes I can do that.

File: src/modules/CSADigitizer/README.md [hunk 1]
```diff
@@ -19,6 +19,8 @@ Alternatively a custom impulse response function can be provided by using the `c
 Noise can be applied to the individual bins of the output pulse, drawn from a normal distribution.
 For the amplified pulse signal, alongside the Time-of-Arrival either the determined Time-over-Threshold, or the integral of the amplified pulse can be stored in the `PixelHit`.
 
+The values stored in `PixelHit` depend on the ToA and ToT settings. If a ToA clock is defined, then `local_time` will be stored in clock cycles, else in time units. If a ToT clock is defined, then `signal` will be the amount of ToT cycles the pulse is above the threshold, else it will be the integral of the amplified pulse.
```
Comment (schuetzepaul) [inline src/modules/CSADigitizer/README.md:22]: I think line 20 can then be removed, as some information is repeated.

Comment (stephanlachnit) [reply src/modules/CSADigitizer/README.md]: Agreed, updated it.

**Comment by schuetzepaul** [comment]
Perfect, thank you very much!

**Pull request merged**